demo/src/com/google/maps/android/utils/demo/KmlDemoActivity.java

```diff
@@ -32,19 +31,48 @@ protected int getLayoutId() {
 
     public void startDemo () {
         try {
-            KmlLayer kmlLayer = new KmlLayer(getMap(), R.raw.point, getApplicationContext());
-            kmlLayer.addLayer();
-            KmlContainer container = kmlLayer.getContainers().iterator().next();
-            KmlPlacemark placemark = container.getPlacemarks().iterator().next();
-            KmlPolygon polygon = (KmlPolygon) placemark.getGeometry();
-            polygon.getOuterBoundaryCoordinates().get(0);
-            getMap().moveCamera(CameraUpdateFactory.newLatLngZoom(
-            polygon.getOuterBoundaryCoordinates().get(0), 18));
+            mMap = getMap();
+            //retrieveFileFromResource();
+            retrieveFileFromUrl();
         } catch (Exception e) {
             Log.e("Exception caught", e.toString());
         }
     }
 
+    private void retrieveFileFromResource() {
+        try {
+            KmlLayer kmlLayer = new KmlLayer(mMap, R.raw.campus, getApplicationContext());
+            kmlLayer.addLayerToMap();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void retrieveFileFromUrl() {
+        String url = "https://kml-samples.googlecode.com/svn/trunk/" +
+                "morekml/Polygons/Polygons.Google_Campus.kml";
+        new DownloadKmlFile(url).execute();
+    }
+
+    private void moveCameraToKml(KmlLayer kmlLayer) {
+        //Retrieve the first container in the KML layer
+        KmlContainer container = kmlLayer.getContainers().iterator().next();
+        //Retrieve a nested container within the first container
+        container = container.getContainers().iterator().next();
+        //Retrieve the first placemark in the nested container
+        KmlPlacemark placemark = container.getPlacemarks().iterator().next();
+        //Retrieve a polygon object in a placemark
+        KmlPolygon polygon = (KmlPolygon) placemark.getGeometry();
+        //Create LatLngBounds of the outer coordinates of the polygon
+        LatLngBounds.Builder builder = new LatLngBounds.Builder();
+        for (LatLng latLng : polygon.getOuterBoundaryCoordinates()) {
+            builder.include(latLng);
+        }
+        getMap().moveCamera(CameraUpdateFactory.newLatLngBounds(builder.build(), 1));
+    }
+
```

Comment:
Demo LGTM!

[![LGTM](http://www.lgtm.in/p/wQgMfYynq)](http://www.lgtm.in/i/wQgMfYynq)